apps/src/applab/designProperties.jsx

```diff
@@ -48,6 +29,7 @@ var DesignProperties = module.exports = React.createClass({
     // TODO (brent) - right now if i create two elements with the same id, I
     // can still run into the same problem, where I click on the other element
     // and the table doesn't update
+    // TODO (brent) - it appears the wrong element sometimes gets deleted
     return (
       <div key={this.props.element.id}>
```

Comment:
Agreed on a different source of key. ordering within parent div will probably also get changed with move forward, move to front, move backward, move to back.